astropy/units/quantity.py

```diff
@@ -783,12 +810,30 @@ def equivalencies(self):
 
         return self._equivalencies
 
+    def _recursively_apply(self, func):
```

Comment:
I like this solution. Very clean. The ``operator.attrgetter`` is neat.

Reply:
I'd actually like this to live on the unit rather than the `Quantity`, as the current `.si` implementation would not work for, say, converting `F` to `C`, but am leaving that for later.